src/kaltura-player.ts

```diff
@@ -186,6 +186,7 @@ export class KalturaPlayer extends FakeEventTarget {
     const sources = Utils.Object.mergeDeep({}, playerConfig.sources, this._localPlayer.sources);
     if (playerConfig.sources.type === MediaType.VOD && this._localPlayer.sources.mediaEntryType?.toString() === MediaType.LIVE) {
       sources.id = playerConfig.sources.id;
+      sources.duration = playerConfig.sources.duration;
```

Comment:
[live streams can have infinite duration](https://github.com/kaltura/playkit-js/blob/1463a414726752aa3b97d54b6d89b49b07d3ea90/src/engines/html5/media-source/adapters/native-adapter.ts#L1265)
why do you need to set the duration ? and why does it need to be on the source when the player has an api to retrieve duration ?
also live duration can change during runtime...

Reply:
it's the same issue like - https://github.com/kaltura/kaltura-player-js/pull/739
we are talking about entries that were live in the past and now it's vod (the condition check this) but the live entryId in use so the sources.duration have the **live** duration but the compare in the timeline will be with the vod duration so need from the beginning use the vod duration

Reply:
got it
since this is the second time this code is updated - do you think there are any additional fields that are missing here ?

Reply:
From my point of view there is nothing obvious, don't want to change if we are not sure.